sci-libs/gsl/gsl-2.8.ebuild

```
@@ -0,0 +1,65 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools flag-o-matic toolchain-funcs
+
+DESCRIPTION="The GNU Scientific Library"
+HOMEPAGE="https://www.gnu.org/software/gsl/"
+SRC_URI="mirror://gnu/${PN}/${P}.tar.gz
+	https://github.com/Jamim/${PN}/commit/${P//./%2E}-cblas.patch -> ${P}-cblas.patch"
```

github commit `.patch` urls are not reliably reproducible unless you do this:

```suggestion
	https://github.com/Jamim/${PN}/commit/${P//./%2E}-cblas.patch?full_index=1 -> ${P}-cblas.patch"
```

---

For those who are wondering what the issue is, there is [a Reddit post](https://www.reddit.com/r/archlinux/comments/1dhtzys/psa_in_pkgbuild_use_the_full_index_url_param_for/) that explains it.

---

It's sort of magic information that everyone copied from Homebrew, probably because Homebrew has github staff maintaining homebrew so they either had inside knowledge about this secret parameter, or implemented it specifically for use in homebrew.

Basically, it causes `git format-patch`, used for producing the backing content of the .patch files on github, to be invoked with `--full-index`. The git manpage explains how that works in greater detail.